### DISCUSSION TITLE
Adjacency matrix to TUDataset format

### DISCUSSION THREAD

**Discussion by saiprasanna06** [pyg-team/pytorch_geometric#6016] [Q&A]
I have a set of 100 graphs with corresponding graph labels. The adjacency matrices are stored in a container as NumPy arrays.  I want to convert them into TUDataset format. Is there any way I can do this?

**Comment by wsad1** [comment]
By TUDataset format do you mean a sparse `edge_index` of shape `(2,num_edges)`. If so, you can convert you adjacency matrix from a numpy array to a torch tensor. If the adjacency matrix is dense and of shape `num_nodes*num_nodes` you can use the [dense_to_sparse](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dense_to_sparse) to convert it to a sparse representation.
